index.js

```diff
@@ -481,97 +499,8 @@ module.exports = {
-          ),
-          pull.drain(
-            (msg) => {
-              const groupId = msg.value.content.recps[0]
-              ssb.box2.excludeGroupInfo(groupId, (err) => {
-                // prettier-ignore
-                if (err) return cb(clarify(err, 'Error on excluding group info after finding exclusion of ourselves'))
-              })
-            },
-            (err) => {
-              // prettier-ignore
-              if (err) return cb(clarify(err, 'Error on looking for exclude messages excluding us'))
-            }
-          )
-        )
-
-        // look for new epochs that we're added to
-        pull(
-          ssb.db.query(
-            where(and(isDecrypted('box2'), type('group/add-member'))),
-            live({ old: true }),
-            toPullStream()
-          ),
-          pull.filter(isAddMember),
-          // groups/epochs we're added to
-          pull.filter((msg) => {
-            return msg.value.content.recps.includes(myRoot.id)
-          }),
-          // to find new epochs we only check groups we've accepted the invite to
-          paraMap((msg, cb) => {
-            pull(
-              ssb.box2.listGroupIds(),
-              pull.filter((groupId) => groupId === msg.value.content.recps[0]),
-              pull.take(1),
-              pull.collect((err, groupIds) => {
-                // prettier-ignore
-                if (err) return cb(clarify(err, "Error getting groups we're already in when looking for new epochs"))
-                cb(null, groupIds.length ? msg : null)
-              })
-            )
-          }, 4),
-          pull.filter(Boolean),
-          pull.drain(
-            (msg) => {
-              const groupId = msg.value.content.recps[0]
-
-              const newKey = Buffer.from(msg.value.content.groupKey, 'base64')
-              ssb.box2.addGroupInfo(groupId, { key: newKey }, (err) => {
-                // prettier-ignore
-                if (err) return cb(clarify(err, 'Error adding new epoch key that we found'))
-
-                const newKeyPick = {
-                  key: newKey,
-                  scheme: keySchemes.private_group,
-                }
-                // TODO: naively guessing that this is the latest key for now
-                ssb.box2.pickGroupWriteKey(groupId, newKeyPick, (err) => {
-                  // prettier-ignore
-                  if (err) return cb(clarify(err, 'Error switching to new epoch key that we found'))
-
-                  ssb.db.reindexEncrypted((err) => {
-                    // prettier-ignore
-                    if (err) cb(clarify(err, 'Error reindexing after finding new epoch'))
-                  })
-                })
-              })
-            },
-            (err) => {
-              // prettier-ignore
-              if (err) return cb(clarify(err, "Error finding new epochs we've been added to"))
-            }
-          )
-        )
+        cb(null)
+        startListeners(ssb, getPreferredEpoch, console.error)
```

Comment:
BOOM, look at this cute lil guy. Tidies up a whole mess.
Also takes an error-handler at the end, so we can easily decide what we wanna do with those errors later and change this in one place

Reply:
woo nice